src/main/java/bio/terra/common/FutureUtils.java

```diff
@@ -74,7 +74,7 @@ private FutureUtils() {}
                       foundFailure.compareAndSet(null, ere);
                     } else {
                       foundFailure.compareAndSet(
-                          null, new ApiException("Error executing thread", e));
+                          null, new ApiException(e.getMessage(), e.getCause()));
```

Comment:
Could any sensitive errors pop up from this refactor?

Reply:
That's a very good question, and is causing me to reconsider this PR.

The existing code already had handling for when the cause of the future's failure was an `ErrorReportException`, meaning our code consciously chose to construct an exception meant for the end user.  In that case, we could be confident that sensitive error information wasn't inadvertently passed along.

But in this case, we'd be passing along an unexpected and unchecked error, so would not have that same confidence.  The more conservative approach is for developers to investigate such errors when they come up, make code changes to handle them gracefully, and massage them into user-facing errors.

If we stick with the more conservative approach, do you have any thoughts on whether we can make the "Error executing thread" message better for the user?

Reply:
I am fine making the change keeping in mind that we may need to roll it back in the future.

Reply:
Sounds good -- thanks!